xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/ReferenceRenamer.java

```diff
@@ -46,6 +49,27 @@ public interface ReferenceRenamer
     boolean renameReferences(Block block, DocumentReference currentDocumentReference, DocumentReference oldTarget,
         DocumentReference newTarget, boolean relative);
 
+    /**
+     * Change references of the given block so that the references pointing to the old target points to the new target.
+     *
+     * @param block the {@link Block} to modify
+     * @param currentDocumentReference the current document reference
+     * @param oldTarget the previous reference of the renamed entity (attachment or document)
+     * @param newTarget the new reference of the renamed entity (attachment or document)
+     * @param relative {@code true} if the link should be serialized relatively to the current document
+     * @param updatedDocuments the list of documents that have been renamed in the same job: this list contains the
+     *                          old references before the rename
+     * @return {@code true} if the given {@link Block} was modified
+     * @since 16.10.0RC1
+     */
+    @Unstable
+    default boolean renameReferences(Block block, DocumentReference currentDocumentReference,
+        DocumentReference oldTarget,
+        DocumentReference newTarget, boolean relative, Set<DocumentReference> updatedDocuments)
```

Comment:
`updatedDocuments` does not seem enough to me. I don't see how you can properly cover all use cases without knowing where those pages are moved.